tasks/sep/032-op-pectra-defence/VALIDATION.md

```diff
@@ -0,0 +1,137 @@
+>
+> ### Security Council
+>
+> - Domain Hash: `0xbe081970e9fc104bd1ea27e375cd21ec7bb1eec56bfe43347c3e36c5d27b8533`
+> - Message Hash: `0xbe081970e9fc104bd1ea27e375cd21ec7bb1eec56bfe43347c3e36c5d27b8533`
+
+## Nested Safe State Overrides and Changes
+
+This task is executed by the nested 2/2 `ProxyAdminOwner` Safe. Refer to the
+[generic nested Safe execution validation document](../../../NESTED-VALIDATION.md)
+for the expected state overrides and changes.
+
+The `approvedHashes` mapping **key** of the `ProxyAdminOwner` that should change during the simulation is
+- Council simulation: `0xe80b66d7e4042fb82e51d81480a1ad1600dd392a4af9f99842b6e1450047fce0`
+- Foundation simulation: `0x63bd348b4f1617eabcac49f67f2f40a9040947fd4dad3f262041c7c65a6efce8`
+
+calculated as explained in the nested validation doc.
+
+Additionally, the nonces [will increment by one](../../../NESTED-VALIDATION.md#nonce-increments).
+
+## State Changes
+
+Note: The changes listed below do not include safe nonce updates or liveness guard related changes. Refer to the
+[generic nested Safe execution validation document](../../../NESTED-VALIDATION.md)
+
+### `0x05F9613aDB30026FFd634f38e5C4dFd30a197Fa1` (`DisputeGameFactoryProxy`)
+
+Click on 'show raw state changes'.
+
+- **Key**: `0x4d5a9bd2e41301728d41c8e705190becb4e74abe869f75bdb405b63716a35f9e` <br/>
+  **Before**: `0x000000000000000000000000bbdbdfe37c02439764de0e41c906e4396b5b3914` <br/>
+  **After**: `0x0000000000000000000000001c3eb0ebd6195ab587e1ded358a87bdf9b56fe04` <br/>
+  **Meaning**: Updates the implementation for game type 1. Verify that the old implementation is set in this slot using
+  `cast call 0x05F9613aDB30026FFd634f38e5C4dFd30a197Fa1 "gameImpls(uint32)(address)" 1`.
+
+- **Key**: `0xffdfc1249c027f9191656349feb0761381bb32c9f557e01f419fd08754bf5a1b` <br/>
+  **Before**: `0x000000000000000000000000f3ccf0c4b51d42cfe6073f0278c19a8d1900e856` <br/>
+  **After**: `0x000000000000000000000000927248cb1255e0f02413a758899db4aecffaa5fe` <br/>
+  **Meaning**: Updates the implementation for game type 0. Verify that the old implementation is set using
+  `cast call 0x05F9613aDB30026FFd634f38e5C4dFd30a197Fa1 "gameImpls(uint32)(address)" 0`.
+
+
+## Verifying Dispute Games
+
+The old and new dispute game contracts can be compared with the [comparegames.sh](https://gist.github.com/ajsutton/28be852a36d9d19af16f7c870b267873)
+script.
+
+From the tenderly simulation click on 'Run on Fork', then copy and export the RPC url provided.
+
+```
+export ETH_RPC_URL=https://rpc.tenderly.co/fork/...
+```
+
+The arguments to the script can be taken from the before and after values in the `DisputeGameFactoryProxy` above.
+
+### PermissionedDisputeGame:
+
+The only change seen here is the `absolutePrestate()` as expected.
+
+```shell
+$ just --justfile ../../../justfile compare-games 0xbbdbdfe37c02439764de0e41c906e4396b5b3914 0x1c3eb0ebd6195ab587e1ded358a87bdf9b56fe04
```

Comment:
I think this does not load the ETH_RPC_URL properly? Looks like its default to localhost for me:

```
$ just --justfile ../../../justfile compare-games 0xbbdbdfe37c02439764de0e41c906e4396b5b3914 0x1c3eb0ebd6195ab587e1ded358a87bdf9b56fe04
bash script/utils/compareGames.sh 0xbbdbdfe37c02439764de0e41c906e4396b5b3914 0x1c3eb0ebd6195ab587e1ded358a87bdf9b56fe04
Error: error sending request for url (http://localhost:8545/)
```